governance/proposals/012-set-protocol-fee-cross-bridge.js

```diff
@@ -0,0 +1,190 @@
+    - unlock : ${unlockAddress}`)
+
+  // get Unlock interface
+  const { interface: unlockInterface } = await ethers.getContractAt(
+    Unlock.abi,
+    unlockAddress
+  )
+
+  // parse unlock calls
+  const calls = [
+    {
+      contractAddress: unlockAddress,
+      calldata: unlockInterface.encodeFunctionData('setProtocolFee', [
+        protocolFee,
+      ]),
+      value: 0,
+      operation: 1, // Unlock is a proxy so use DELEGATECALL
+    },
+  ]
+
+  const oldProtocolFee = await unlockInterface.protocolFee()
+  const explainers = [
+    `- Protocol fee for ${unlockAddress} set to ${PROTOCOL_FEE_STRING}
+  \`setProtocolFee(${protocolFee})\` (Previous Protocol Fee: ${ethers.formatEther(
+    oldProtocolFee
+  )})`,
+  ]
+
+  return {
+    calls,
+    explainers,
+  }
+}
+
+const parseSafeCall = async ({ destChainId, calls }) => {
+  // parse multicall
+  const { to, data, value, operation } = await parseSafeMulticall({
+    chainId: destChainId,
+    calls,
+  })
+
+  // encode multicall instructions to be executed by the SAFE
+  const abiCoder = ethers.AbiCoder.defaultAbiCoder()
+  const moduleData = abiCoder.encode(
+    ['address', 'uint256', 'bytes', 'bool'],
+    [
+      to, // to
+      value, // value
+      data, // data
+      operation, // operation: 0 for CALL, 1 for DELEGATECALL
+    ]
+  )
+  return moduleData
+}
+
+const parseBridgeCall = async ({ destChainId, moduleData }) => {
+  const { governanceBridge } = await getNetwork(destChainId)
+
+  // get bridge info on receiving chain
+  const {
+    domainId: destDomainId,
+    modules: { connextMod: destAddress },
+  } = governanceBridge
+
+  // get bridge address on mainnet
+  const {
+    governanceBridge: { connext: bridgeAddress },
+  } = await getNetwork(1)
+
+  if (!destDomainId || !destAddress) {
+    throw Error('Missing bridge information')
+  }
+
+  // parse call for bridge
+  return {
+    contractAddress: bridgeAddress,
+    contractNameOrAbi: IConnext,
+    functionName: 'xcall',
+    functionArgs: [
+      destDomainId,
+      destAddress, // destMultisigAddress,
+      ADDRESS_ZERO, // asset
+      ADDRESS_ZERO, // delegate
+      0, // amount
+      30, // slippage
+      moduleData, // calldata
+    ],
+  }
+}
+
+module.exports = async () => {
+  const explainers = []
+
+  // get oracle calls for mainnet
+  console.log(`Parsing for Ethereum Mainnet (1)`)
+  const { calls: mainnetCalls, explainers: mainnetExplainers } =
+    await parseSetProtocolFeeCalls(1)
+
+  explainers.push({
+    name: 'Ethereum Mainnet',
+    id: 1,
+    explainers: mainnetExplainers,
+  })
+
+  // get setProtocolFee call data for dest chains
+  const bridgedCalls = []
+  for (let i in [...targetChains]) {
+    const { name, id } = targetChains[i]
+    console.log(`Parsing for chain ${name} (${id})`)
+    const { calls, explainers: destExplainers } =
+      await parseSetProtocolFeeCalls(id)
+    const moduleData = await parseSafeCall({ destChainId: id, calls })
```

Comment:
Maybe that step can be skipped as it was intended mostly to pack multiple calls into a single Safe multicall. Here there is only 1 call on each chain so the ˋparseSetProtocolFeeˋ function could directly returned the correct module data for Safe.